### PR TITLE
Prepare package for rc-2.1.5

### PR DESCRIPTION
**Pull request by CJ42**
# What does this PR introduce?

**Prepare release candidate 2.1.5**

## Optimise npm package size from 11MB to 550kB 🚀  📦 

**Files included in npm package**
- [x] include only `ERC725` `ILSP` contracts artifacts (not artifacts from `node_modules`)
- [x] do not include `helpers/` Solidity contracts

## Other changes

- [x] removed unused npm package from devDependencies (`@openzeppelin/cli`, `ethereumjs-tx/utils`)
- [x] created **nodejs file `small-artifacts.js`** to remove unnecessary entries in JSON artifacts and reduce drastically artifacts file size. 
- [x] added npm script `package` to execute `small-artifacts.js` and optimise artifacts before publishing to npm

### CODE REVIEW

**Comment by Hugoo** [comment]
Nice !

Would be cool if `truffle compile` had an option so we can tell it to not put these extra keys in the generated JSON.

Also, I'm not sure if this PR is a RC as I don't see any tag or version bump. Only new features. May makes sense to rename to a more descriptive PR title imo.

**Review by Hugoo — APPROVED** [review]

File: implementations/package.json [hunk 1]
```diff
@@ -21,7 +25,8 @@
     "lint": "solhint 'contracts/**/*.sol'",
     "test": "truffle test",
     "solc": "sh solc.sh",
-    "publish": "truffle compile --all; npm run test; npm publish"
+    "publish": "truffle compile --all; npm run test; npm run package; npm publish",
+    "package": "node optimize-artifacts.js"
```
Comment (frozeman) [inline implementations/package.json:29]: reduce-abi is better

**Review by frozeman — CHANGES_REQUESTED** [review]

File: implementations/package.json (hunk 1 above)
Comment (CJ42) [reply implementations/package.json:29]: @frozeman I would disagree here, as the abi is one among many entries in the JSON file.
But in the end, it is just a filename 🙂

Comment (frozeman) [reply implementations/package.json:29]: reduce-artifacts

Comment (CJ42) [reply implementations/package.json]: Ah the joy of finding good and meaningful file names! 😄  Haha
Done @frozeman ✅

**Pull request merged**